### PR TITLE
SYNKRON-276  Added Update Inventory 

### PR DESCRIPTION
**Pull request by Hardik2023103**
create the update stock method by post request

### CODE REVIEW

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class updateStockResponse {
+    @JsonProperty("item")
+    private Item item;
+    private Integer stockCount;
+    private Double quantity;
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: change datatype Double to Integer

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class updateStockResponse {
+    @JsonProperty("item")
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: why need @JsonProperty ??

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class updateStockResponse {
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: change name updateStockResonse to "InventoryResponse"
**Note: class name always First letter capital**

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package io.github.dftrakesh.cloverrest.model.updateInventory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class updateRequest {
+    /*private String id;*/
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java]: remove comment code

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package io.github.dftrakesh.cloverrest.model.updateInventory;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Data;
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class updateRequest {
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java]: change class name updateRequest to InventoryRequest

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/Item.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package io.github.dftrakesh.cloverrest.model.updateInventory;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Data;
+import java.util.List;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Item {
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/Item.java]: already you created item class, so please remove it

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import static io.github.dftrakesh.cloverrest.constantcode.ConstantCodes.V3_MERCHANTS;
+
+public class CloverRestUpdateInventory extends CloverRestSDK {
+
+    public CloverRestUpdateInventory(AccessCredentials accessCredentials) { super(accessCredentials);}
+
+    @SneakyThrows
+    public updateStockResponse update(updateRequest inventoryRequest , String itemId ) {
+        String sUri ="https://sandbox.dev.clover.com/" + V3_MERCHANTS + accessCredentials.getMerchantId() + V3_ITEM_STOCKS_ENDPOINT + "/" + itemId;
+        URI uri = new URI(sUri);
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: one space add

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import static io.github.dftrakesh.cloverrest.constantcode.ConstantCodes.V3_ITEM_STOCKS_ENDPOINT;
+import static io.github.dftrakesh.cloverrest.constantcode.ConstantCodes.V3_MERCHANTS;
+
+public class CloverRestUpdateInventory extends CloverRestSDK {
+
+    public CloverRestUpdateInventory(AccessCredentials accessCredentials) { super(accessCredentials);}
+
+    @SneakyThrows
+    public updateStockResponse update(updateRequest inventoryRequest , String itemId ) {
+        String sUri ="https://sandbox.dev.clover.com/" + V3_MERCHANTS + accessCredentials.getMerchantId() + V3_ITEM_STOCKS_ENDPOINT + "/" + itemId;
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: remove sandbox url and relevant url add

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import java.net.http.HttpResponse;
+import static io.github.dftrakesh.cloverrest.constantcode.ConstantCodes.V3_ITEM_STOCKS_ENDPOINT;
+import static io.github.dftrakesh.cloverrest.constantcode.ConstantCodes.V3_MERCHANTS;
+
+public class CloverRestUpdateInventory extends CloverRestSDK {
+
+    public CloverRestUpdateInventory(AccessCredentials accessCredentials) { super(accessCredentials);}
+
+    @SneakyThrows
+    public updateStockResponse update(updateRequest inventoryRequest , String itemId ) {
```
Comment (DhruvFefar2001) [inline src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: update method move to CloverRestItem Class and change method name update to "updateInventory"

**Review by DhruvFefar2001 — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java (hunk 9 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: Moved and changed

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java (hunk 8 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: removed and added relevant url

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java (hunk 7 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/CloverRestUpdateInventory.java]: added one space

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/Item.java (hunk 6 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/Item.java]: Deleted Item Class

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java (hunk 5 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java]: Updated name

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java (hunk 4 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateRequest.java]: removed comment

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java (hunk 3 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: changed  name

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java (hunk 2 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: removed @JsonProperty

File: src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java (hunk 1 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/model/updateInventory/updateStockResponse.java]: In response, i got quantity in double
So not changing

**Review by DhruvFefar2001 — APPROVED** [review]

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestItem.java [hunk 10]
```diff
@@ -33,4 +37,14 @@ public ItemElement getItem(HashMap<String, String> param, String itemId) {
         HttpResponse.BodyHandler<ItemElement> handler = new JsonBodyHandler<>(ItemElement.class);
         return getRequestWrapped(request, handler);
     }
+
+    @SneakyThrows
+    public InventoryResponse updateInventory(InventoryRequest inventoryRequest , String itemId ) {
```
Comment (dftrakesh) [inline src/main/java/io/github/dftrakesh/cloverrest/CloverRestItem.java]: Format code properly.

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestSDK.java [hunk 11]
```diff
@@ -50,6 +53,16 @@ protected HttpRequest get(URI uri) {
                           .build();
     }
 
+    @SneakyThrows
+    protected HttpRequest post(URI uri, final Object object) {
+        String jsonBody = objectMapper.writeValueAsString(object);
+        return HttpRequest.newBuilder(uri)
+            .header(AUTHORIZATION, "Bearer " + this.accessCredentials.getAccessToken())
```
Comment (dftrakesh) [inline src/main/java/io/github/dftrakesh/cloverrest/CloverRestSDK.java]: Remove this.

**Review by dftrakesh — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/dftrakesh/cloverrest/CloverRestSDK.java (hunk 11 above)
Comment (Hardik2023103) [reply src/main/java/io/github/dftrakesh/cloverrest/CloverRestSDK.java]: Removed "this" keyword

**Pull request merged**